src/analytics/internal/AnalyticsStorage.js

```diff
@@ -0,0 +1,12 @@
+import AnalyticsLogStorage from './AnalyticsLogStorage'
+import AnalyticsStatisticsStorage from './AnalyticsStatisticsStorage'
+
+export default async function storeEventAndStatistics({ event, statistics }) {
+    // Process data to store the data
+    await AnalyticsLogStorage(event)
+
+    if (statistics) {
+        const statistic = new AnalyticsStatisticsStorage(statistics.notifType)
```

Comment:
I don't see the point in using classes for all these modules if a new instance is made each time the storage function gets called. And some modules are written as classes, like `AnalyticsStatisticsStorage`, but other modules that are seemingly for a similar purpose are just functions, like `AnalyticsLogStorage` (and no PascalCase for function names! although to be fair this should be set as a linting rule).

Why not make this `AnalyticsStorage` module export a class that you can instantiate and has `storeEvent` as a method? I don't see any point to the statistics storage module, as it's not really "storing" anything; it simply tells the main internal `Analytics` instance to update its in-memory statistics. If you draw out how this flows between the modules on paper you will see a big loop from: internal analytics -> storage -> stats storage -> count statistics -> internal analytics. Try to avoid these.